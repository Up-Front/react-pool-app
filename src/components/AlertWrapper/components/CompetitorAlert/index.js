import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { BounceIn, BounceOut } from 'animate-css-styled-components';
import {
  AlertWrapper,
  WinnerWarningCenter,
  BoomText,
  BoomTextShadow,
} from './styles';
import Avatar from './components/Avatar';

class CompetitorAlert extends Component {
  componentDidMount() {
    this.animationRef.addEventListener('animationend', this.animationDone);
  }
  componentWillUnmount() {
    this.animationRef.removeEventListener('animationend', this.animationDone);
  }

  animationDone = () => {
    console.log('animation done');
    this.props.onEndAnimation();
  };

  calculateStreak = gameList => {
    let streak = 1;
    gameList.split('').reduceRight((prev, current) => {
      if (current === prev) {
        streak++;
      }
      return prev;
    });
    return streak;
  };

  setVictoryName = streak => {
    if (streak >= 10) {
      return 'godlike';
    } else if (streak >= 6) {
      return 'rampage';
    } else if (streak >= 3) {
      return 'dominating';
    }
    return 'victory';
  };

  victoryName = 'Game on!';
  // let victoryName = this.setVictoryName(
  //   this.calculateStreak()
  // );
  // <audio
  //         autoPlay="autoplay"
  //         src={'/sound/victory/' + victoryName + '.mp3'}
  //         type="audio/mpeg"
  //       />

  render() {
    return (
      <AlertWrapper>
        <WinnerWarningCenter>
          <BounceIn delay=".5s" duration="1s">
            <BounceOut
              delay="5s"
              duration=".5s"
              innerRef={x => {
                this.animationRef = x;
              }}
            >
              <Avatar user={this.props.opponent} />
            </BounceOut>
          </BounceIn>
          <BounceIn delay=".7s" duration=".5s">
            <BounceOut
              delay="5s"
              duration=".3s"
              innerRef={x => {
                this.animationRef = x;
              }}
            >
              <BoomText>{this.victoryName}!</BoomText>
              <BoomTextShadow>{this.victoryName}!</BoomTextShadow>
            </BounceOut>
          </BounceIn>
        </WinnerWarningCenter>
      </AlertWrapper>
    );
  }
}

CompetitorAlert.propTypes = {
  opponent: PropTypes.object,
};

export default CompetitorAlert;
