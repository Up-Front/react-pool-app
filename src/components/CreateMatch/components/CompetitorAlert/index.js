import React from 'react';
import PropTypes from 'prop-types';

const CompetitorAlert = ({ auth }) => {
  return <div>dd</div>;
};

CompetitorAlert.propTypes = {
  auth: PropTypes.object.isRequired,
};

export default CompetitorAlert;
