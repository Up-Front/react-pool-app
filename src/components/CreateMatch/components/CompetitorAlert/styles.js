import styled, { injectGlobal } from 'styled-components';

injectGlobal`
  @font-face {
    font-family: 'Permanent Marker';
    font-style: normal;
    font-weight: 400;
    src: local('Permanent Marker Regular'), local('PermanentMarker-Regular'), url(https://fonts.gstatic.com/s/permanentmarker/v7/Fh4uPib9Iyv2ucM6pGQMWimMp004La2Cf5b6jlg.woff2) format('woff2');
    unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
    }
`;

export const AlertWrapper = styled.div`
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  z-index: 1000;

  & > div {
    display: flex;
    align-items: center;
    justify-content: center;

    & > div {
      flex: 1;
      position: absolute;
    }

    & > div:last-child {
      width: 100vw;
    }
  }
`;

export const WinnerWarningCenter = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100vw;
  height: 30vw;
`;

export const AvatarImage = styled.img`
  position: absolute;
  top: 0;
  width: 30vw;
  height: 30vw;
  border-radius: 50%;
`;

export const LeaderIcon = styled.div`
  position: relative;
  width: 15vw;
  height: 15vw;
  transform: rotate(-30deg);

  &:after {
    content: 'ðŸ‘‘';
    position: relative;
    top: -5vw;
    left: 5vw;
    font-size: 8vw;
  }
`;

export const BoomTextBase = styled.h2`
  position: absolute;
  top: -7vw;
  transform: rotate(-5deg);
  width: 100vw;
  text-align: center;
  font-family: 'Permanent Marker';
  font-size: 10vw;
`;

export const BoomText = styled(BoomTextBase)`
  background: linear-gradient(
    to bottom,
    #e8bf2e 0%,
    #e58500 100%
  ); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  z-index: 1;
`;

export const BoomTextShadow = styled(BoomTextBase)`
  opacity: 0.4;
  text-shadow: 4px 4px 2px rgba(0, 0, 0, 1);
`;
