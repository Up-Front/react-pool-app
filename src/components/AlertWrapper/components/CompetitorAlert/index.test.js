import React from 'react';
import { shallow } from 'enzyme';
import CompetitorAlert from './';

describe('CompetitorAlert component', () => {
  test('renders without crashing', () => {
    const auth = {
      uid: 'abc',
      displayName: 'He-man',
    };

    const match = {
      matchId: 'xyz',
      competitors: [
        {
          uid: 'abc',
          displayName: 'He-man',
        },
        {
          uid: 'xyz',
          displayName: 'skeletor',
        },
      ],
    };
    const wrapper = shallow(<CompetitorAlert auth={auth} match={match} />);
    expect(wrapper).toMatchSnapshot();
  });
});
