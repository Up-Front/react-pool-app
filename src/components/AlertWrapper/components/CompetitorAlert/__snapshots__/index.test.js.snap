// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompetitorAlert component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompetitorAlert
    auth={
        Object {
            "displayName": "He-man",
            "uid": "abc",
          }
    }
    match={
        Object {
            "competitors": Array [
              Object {
                "displayName": "He-man",
                "uid": "abc",
              },
              Object {
                "displayName": "skeletor",
                "uid": "xyz",
              },
            ],
            "matchId": "xyz",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        undefined,
        <styled.div>
          <Styled(styled.div)
                    delay=".5s"
                    duration="1s"
          >
                    <Styled(styled.div)
                              delay="5s"
                              duration=".5s"
                    >
                              <Avatar
                                        user={
                                                  Object {
                                                            "displayName": "skeletor",
                                                            "uid": "xyz",
                                                          }
                                        }
                              />
                    </Styled(styled.div)>
          </Styled(styled.div)>
          <Styled(styled.div)
                    delay=".7s"
                    duration=".5s"
          >
                    <Styled(styled.div)
                              delay="5s"
                              duration=".3s"
                              innerRef={[Function]}
                    >
                              <Styled(styled.h2)>
                                        Game on
                                        !
                              </Styled(styled.h2)>
                              <Styled(styled.h2)>
                                        Game on
                                        !
                              </Styled(styled.h2)>
                    </Styled(styled.div)>
          </Styled(styled.div)>
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(styled.div)
              delay=".5s"
              duration="1s"
>
              <Styled(styled.div)
                            delay="5s"
                            duration=".5s"
              >
                            <Avatar
                                          user={
                                                        Object {
                                                                      "displayName": "skeletor",
                                                                      "uid": "xyz",
                                                                    }
                                          }
                            />
              </Styled(styled.div)>
</Styled(styled.div)>,
            <Styled(styled.div)
              delay=".7s"
              duration=".5s"
>
              <Styled(styled.div)
                            delay="5s"
                            duration=".3s"
                            innerRef={[Function]}
              >
                            <Styled(styled.h2)>
                                          Game on
                                          !
                            </Styled(styled.h2)>
                            <Styled(styled.h2)>
                                          Game on
                                          !
                            </Styled(styled.h2)>
              </Styled(styled.div)>
</Styled(styled.div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(styled.div)
                delay="5s"
                duration=".5s"
>
                <Avatar
                                user={
                                                Object {
                                                                "displayName": "skeletor",
                                                                "uid": "xyz",
                                                              }
                                }
                />
</Styled(styled.div)>,
              "delay": ".5s",
              "duration": "1s",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Avatar
                  user={
                                    Object {
                                                      "displayName": "skeletor",
                                                      "uid": "xyz",
                                                    }
                  }
/>,
                "delay": "5s",
                "duration": ".5s",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "user": Object {
                    "displayName": "skeletor",
                    "uid": "xyz",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(styled.div)
                delay="5s"
                duration=".3s"
                innerRef={[Function]}
>
                <Styled(styled.h2)>
                                Game on
                                !
                </Styled(styled.h2)>
                <Styled(styled.h2)>
                                Game on
                                !
                </Styled(styled.h2)>
</Styled(styled.div)>,
              "delay": ".7s",
              "duration": ".5s",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Styled(styled.h2)>
                    Game on
                    !
</Styled(styled.h2)>,
                  <Styled(styled.h2)>
                    Game on
                    !
</Styled(styled.h2)>,
                ],
                "delay": "5s",
                "duration": ".3s",
                "innerRef": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "Game on",
                      "!",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Game on",
                    "!",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "Game on",
                      "!",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Game on",
                    "!",
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          undefined,
          <styled.div>
            <Styled(styled.div)
                        delay=".5s"
                        duration="1s"
            >
                        <Styled(styled.div)
                                    delay="5s"
                                    duration=".5s"
                        >
                                    <Avatar
                                                user={
                                                            Object {
                                                                        "displayName": "skeletor",
                                                                        "uid": "xyz",
                                                                      }
                                                }
                                    />
                        </Styled(styled.div)>
            </Styled(styled.div)>
            <Styled(styled.div)
                        delay=".7s"
                        duration=".5s"
            >
                        <Styled(styled.div)
                                    delay="5s"
                                    duration=".3s"
                                    innerRef={[Function]}
                        >
                                    <Styled(styled.h2)>
                                                Game on
                                                !
                                    </Styled(styled.h2)>
                                    <Styled(styled.h2)>
                                                Game on
                                                !
                                    </Styled(styled.h2)>
                        </Styled(styled.div)>
            </Styled(styled.div)>
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(styled.div)
                delay=".5s"
                duration="1s"
>
                <Styled(styled.div)
                                delay="5s"
                                duration=".5s"
                >
                                <Avatar
                                                user={
                                                                Object {
                                                                                "displayName": "skeletor",
                                                                                "uid": "xyz",
                                                                              }
                                                }
                                />
                </Styled(styled.div)>
</Styled(styled.div)>,
              <Styled(styled.div)
                delay=".7s"
                duration=".5s"
>
                <Styled(styled.div)
                                delay="5s"
                                duration=".3s"
                                innerRef={[Function]}
                >
                                <Styled(styled.h2)>
                                                Game on
                                                !
                                </Styled(styled.h2)>
                                <Styled(styled.h2)>
                                                Game on
                                                !
                                </Styled(styled.h2)>
                </Styled(styled.div)>
</Styled(styled.div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(styled.div)
                  delay="5s"
                  duration=".5s"
>
                  <Avatar
                                    user={
                                                      Object {
                                                                        "displayName": "skeletor",
                                                                        "uid": "xyz",
                                                                      }
                                    }
                  />
</Styled(styled.div)>,
                "delay": ".5s",
                "duration": "1s",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Avatar
                    user={
                                        Object {
                                                            "displayName": "skeletor",
                                                            "uid": "xyz",
                                                          }
                    }
/>,
                  "delay": "5s",
                  "duration": ".5s",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "user": Object {
                      "displayName": "skeletor",
                      "uid": "xyz",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(styled.div)
                  delay="5s"
                  duration=".3s"
                  innerRef={[Function]}
>
                  <Styled(styled.h2)>
                                    Game on
                                    !
                  </Styled(styled.h2)>
                  <Styled(styled.h2)>
                                    Game on
                                    !
                  </Styled(styled.h2)>
</Styled(styled.div)>,
                "delay": ".7s",
                "duration": ".5s",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Styled(styled.h2)>
                      Game on
                      !
</Styled(styled.h2)>,
                    <Styled(styled.h2)>
                      Game on
                      !
</Styled(styled.h2)>,
                  ],
                  "delay": "5s",
                  "duration": ".3s",
                  "innerRef": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        "Game on",
                        "!",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Game on",
                      "!",
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        "Game on",
                        "!",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Game on",
                      "!",
                    ],
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
